core/server/master/src/main/java/alluxio/master/metrics/MetricsMaster.java

```diff
@@ -22,13 +21,25 @@
  * clients.
  */
 public interface MetricsMaster extends Master  {
+  /**
+   * Handles the client's heartbeat request for metrics collection.
+   *
+   * @param clientId the client id
+   * @param hostname the client hostname
+   * @param metrics client-side metrics
+   */
+  void clientHeartbeat(String clientId, String hostname, List<Metric> metrics);
```

Comment:
Since `clientId` is in `metrics`, we could make this one API. If we keep multiple APIs, could we name them symmetrically?

Reply:
What do you mean? This API only handles heartbeat from the client, while the worker heartbeats are handled in the `BlockMaster`.

Reply:
I meant the `clientHeartbeat` and `putWorkerMetrics` methods

Reply:
Got it.